VivistaServer/Templates/dashboard.liquid

```diff
@@ -1,100 +1,260 @@
+						<td>{{users}}</td>
+						<td>{{videos}}</td>
+						<td>{{downloads}}</td>
+					</tr>
+				</tbody>
+			</table>
+		</div>
+		<br>
 	</div>
 </div>
 
 <script src="https://cdn.jsdelivr.net/npm/chart.js"></script>
 <script src="https://cdn.jsdelivr.net/npm/chartjs-adapter-date-fns/dist/chartjs-adapter-date-fns.bundle.min.js"></script>
-
-<style type="text/css">
-	.chartBox {
-		width: 1000px;
-	}
-</style>
 <script>
-	//------Retrieve and initialize data
-	let minuteData = {{ minuteData | json}};
-	let hourData = {{ hourData | json}};
-	let dayData = {{ dayData | json}};
+	let dataOfDay = {{ minuteData | json}};
+	let dataOfWeek = {{ hourData | json}};
+	let dataOfMonth = {{ dayData | json}};
 	let endpoints = {{ endpoints | json }};
-	let selectedEndpoint = endpoints[0];
-	console.log(dayData);
+	let outliers = {{ outliers | json}}
+	let selectedOutliers = [];
 
+	for (let i = 0; i < outliers.length; i++) {
+		outliers[i].reqinfo = JSON.parse(outliers[i].reqinfo);
+	}
+
+	let outlierRows = [];
+	outliers.forEach(o => {
+		let row = document.createElement('tr');
+		let tdRequestTime = document.createElement('td');
+		tdRequestTime.innerHTML = o.seconds * 1000;
+		let tdTimestamp = document.createElement('td');
+		let dateTime = new Date(o.timestamp);
+		let formatedDateTime = (dateTime.getDate()) + '-' + (dateTime.getMonth() + 1) + '-' + dateTime.getFullYear() + ' ' + dateTime.getHours() + ":" + dateTime.getMinutes() + ":" + dateTime.getSeconds();
+
+		tdTimestamp.innerHTML = formatedDateTime.toString();
+		let tdEndpoint = document.createElement('td');
+		tdEndpoint.innerHTML = o.endpoint;
+
+		row.appendChild(tdRequestTime);
+		row.appendChild(tdTimestamp);
+		row.appendChild(tdEndpoint);
+		outlierRows.push(row);
+	});
 
+	//Pagination
+	let index = 0;
+	let allOutliersTable = document.getElementById("allOutliersTable");
+	let alOutliersTableBody = document.getElementById("allOutliersTableBody");
+	NextPage();
+	function NextPage() {
+		if (index < outliers.length) {
+			let count = allOutliersTable.rows.length;
+			for (let i = 1; i < count; i++) {
+				allOutliersTable.deleteRow(1);
+			}
+
+			let max = 0;
+			if ((index + 5) > outliers.length) {
+				max = outliers.length;
+			}
+			else {
+				max = index + 5;
+			}
+			InsertRows(max);
+		}
+	}
+	function PreviousPage() {
+		if ((index - 5) > 0) {
+			let count = allOutliersTable.rows.length;
+			for (let i = 1; i < count; i++) {
+				allOutliersTable.deleteRow(1);
+			}
+			//Note(Tom): Has one row of th
+			count = count - 1;
+
+			index = index - count - 5;
+			let max = index + 5;
+			InsertRows(max);
+		}
+	}
+	function InsertRows(max) {
+		let indexInsert = 0;
+		for (let i = index; i < max; i++) {
+			allOutliersTable.appendChild(outlierRows[i]);
+			indexInsert++;
+		}
+		index = max;
+	}
+
+
+
+	let generalDataOfDay = {{ generalMinuteData | json}};
+	let generalDataOfWeek = {{ generalDataOfWeek | json}};
+	let generalDataOfMonth = {{ generalDataOfMonth | json}};
```

Comment:
Ah deze vul je in met data vanuit de template. Dat zal dan wel hier moeten blijven.

Reply:
Mag ik dan heel dat script laten staan?

Reply:
Zal afhangen van wat de regels rond scopes zijn voor `let`. Als `let` niet over meerdere scripts geen werkt zal je het hele script hier moeten laten staan.